detox/src/realms/DetoxPrimaryContext.test.js

```diff
@@ -345,96 +357,73 @@ describe('DetoxPrimaryContext', () => {
-    const MockedDetoxLogFinalizer = jest.createMockFromModule('../logger/utils/DetoxLogFinalizer');
-    jest.mock('../logger', () => ({
-      DetoxLogger: MockedDetoxLogger,
-      DetoxLogFinalizer: MockedDetoxLogFinalizer,
-      installLegacyTracerInterface: jest.fn(),
-    }));
-    logger = require('../logger');
+    jest.mock('../logger');
+    logger = jest.requireMock('../logger');
 
     jest.mock('../devices/DeviceRegistry');
-    const DeviceRegistry = require('../devices/DeviceRegistry');
+    const DeviceRegistry = jest.requireMock('../devices/DeviceRegistry');
     deviceRegistryIOS = new DeviceRegistry();
     DeviceRegistry.forIOS.mockReturnValue(deviceRegistryIOS);
 
     deviceRegistryAndroid = new DeviceRegistry();
     DeviceRegistry.forAndroid.mockReturnValue(deviceRegistryAndroid);
 
     jest.mock('../devices/allocation/drivers/android/genycloud/GenyDeviceRegistryFactory');
-    const genycloudDeviceRegistryFactory = require('../devices/allocation/drivers/android/genycloud/GenyDeviceRegistryFactory');
+    const genycloudDeviceRegistryFactory = jest.requireMock('../devices/allocation/drivers/android/genycloud/GenyDeviceRegistryFactory');
     deviceRegistryGenyCloud = new DeviceRegistry();
     genycloudDeviceRegistryFactory.forGlobalShutdown.mockReturnValue(deviceRegistryGenyCloud);
 
     // The mocking complexity here is higher than the norm so as to allow for interacting
     // with both the class and the generated instance as mocks; With the latter - even
     // before its creation by the tested-unit (i.e. in its init()).
-    const _IPCServer = jest.createMockFromModule('../ipc/IPCServer');
-    const mockIpcServer = ipcServer = new _IPCServer();
-    const MockIpcServer = IPCServer = jest.fn().mockImplementation(() => { return mockIpcServer; });
-    jest.mock('../ipc/IPCServer', () => MockIpcServer);
-
-    globalLifecycleHandler = {
-      globalInit: jest.fn(),
-      emergencyCleanup: jest.fn(),
-      globalCleanup: jest.fn(),
-    };
+    jest.mock('../ipc/IPCServer');
+    IPCServer = jest.requireMock('../ipc/IPCServer');
+
+    const GenyGlobalLifecycleHandler = jest.createMockFromModule('../devices/lifecycle/GenyGlobalLifecycleHandler');
+    globalLifecycleHandler = new GenyGlobalLifecycleHandler();
+
     jest.mock('../environmentFactory');
-    environmentFactory = require('../environmentFactory');
+    environmentFactory = jest.requireMock('../environmentFactory');
     environmentFactory.createGlobalLifecycleHandler.mockReturnValue(globalLifecycleHandler);
 
     jest.mock('../server/DetoxServer');
-    DetoxServer = require('../server/DetoxServer');
+    DetoxServer = jest.requireMock('../server/DetoxServer');
 
     jest.mock('../DetoxWorker');
+    DetoxWorker = jest.requireMock('../DetoxWorker');
   }
-});
 
-const detoxConfigBase = Object.freeze({
-  behavior: {
-    init: {
-      keepLockFile: false,
+  class DetoxConfigDriver {
```

Comment:
It is an interesting approach with drivers. Maybe we should use it even more, at least, in some test suites.

Reply:
I agree